src/poetry/console/commands/show.py

```diff
@@ -528,7 +528,7 @@ def find_latest_package(
                     return provider.search_for_direct_origin_dependency(dep)
 
         name = package.name
-        selector = VersionSelector(self.poetry.pool)
+        selector = VersionSelector(self.poetry.pool.repositories[0])
```

Comment:
What would be the right way to handle this? Moving from Pool to AbstractRepository.